atxm/machine.py

```diff
@@ -118,40 +153,67 @@ def _handle_errors(self, *args, **kwargs):
         self.log.warn("[recovery] restarting transaction machine!")
         self._start(now=False)
 
-    def _cycle(self) -> None:
-        """Execute one cycle"""
+    # State Transitions
+    @_transition_to_paused.before
+    def _enter_pause_mode(self):
+        self.log.info("[pause] pause mode activated")
+        return
+
+    @_PAUSED.enter
+    def _process_paused(self):
+        # TODO what action should be taken to check if we leave the pause state?
+        #  Perhaps a call to self.__strategize()
```

Comment:
>         #  Perhaps a call to self.__strategize()

I think this is the best approach, more specifically, we can ensure strategize is always called regardless of the current state.  Strategies can assess the need for a pause on each individual iteration.  It's possible that pause is overbuilt - it may not need to be an attribute at all, if it's not useful.

Reply:
Yep. One possibility with the state machine is that it allowsfor busy and pause states to each do their own thing. While in paused state, we could set it up that `_strategize()` is the only thing called, and the state will remain paused until `__strategize`  eventually calls `resume()` internally.